nemo/lightning/fabric/plugins.py

```diff
@@ -52,64 +108,45 @@ def convert_output(self, data: AnyT) -> AnyT:
         """
         return data
 
-    def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
-        from nemo.core.optim import MainParamsOptimizerWrapper
+    def convert_config(self, config: ConfigT) -> ConfigT:
+        """Convert the config to the precision type this plugin handles.
 
-        return MainParamsOptimizerWrapper(
-            optimizer,
-            # https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L496
-            fp32_grad_accum=True,
-            contiguous_grad_bucket=True,
-        )
+        This is optional and depends on the precision limitations during optimization.
+
+        """
+        return update_config_with_dtype_overrides(self.dtype_config, config)
 
     def convert_module(self, module: nn.Module) -> nn.Module:
         """Convert the module parameters to the precision type this plugin handles.
 
         This is optional and depends on the precision limitations during optimization.
 
         """
-        if not hasattr(module, "module"):
-            return module
-
         from megatron.core.transformer.module import Float16Module
         from megatron.core.utils import get_model_config
 
-        if self.precision in ["16-mixed", "bf16-mixed"]:
+        if self.dtype_config.fp16 or self.dtype_config.bf16:
+            # Patch config options
             config = get_model_config(module.module)
```

Comment:
should this line be guarded by `if hasattr(module, 'module'):`?

Reply:
Yeah, I realized that as well. The fix is part of this [PR](https://github.com/NVIDIA/NeMo/pull/10879) that was just merged.